### PR TITLE
OCLOMRS-470: Make “Narrower-Than” the default relationship for external mappings, and “SAME-AS” the default for CIEL (when adding a mapping)

### PR DESCRIPTION
**Pull request by judeinno**
# JIRA TICKET NAME:
[Make “Narrower-Than” the default relationship for external mappings, and “SAME-AS” the default for CIEL (when adding a mapping)](https://issues.openmrs.org/browse/OCLOMRS-470)

# Summary:
Make “Narrower-Than” the default relationship for external mappings (all mappings except CIEL) when adding or editing a concept mapping.

### CODE REVIEW

File: src/components/dictionaryConcepts/components/MapType.jsx [hunk 1]
```diff
@@ -3,12 +3,12 @@ import PropTypes from 'prop-types';
...
 const mapType = (props) => {
   const {
-    index, map_type, map_types, updateEventListener, url,
+    index, map_type, map_types, updateEventListener, url, source,
   } = props;
   return (
     <select
       tabIndex={index}
-      defaultValue={map_type}
+      value={source !== 'CIEL' ? 'NARROWER-THAN' : map_type}
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/components/MapType.jsx]: Shouldn't this be already in a constant?

Comment (judeinno) [reply src/components/dictionaryConcepts/components/MapType.jsx]: I have fixed it and added the constant `INTERNAL_MAPPING_DEFAULT_SOURCE` used for `'CIEL'`

File: src/components/dictionaryConcepts/components/MapType.jsx [hunk 2]
```diff
@@ -33,12 +34,14 @@ mapType.propTypes = {
...
   map_types: PropTypes.array,
   url: PropTypes.string,
+  source: PropTypes.string,
 };
 
 mapType.defaultProps = {
   url: '',
   map_type: '',
   index: 0,
+  source: 'CIEL',
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/components/MapType.jsx]: Don't we already have a constant for that?

File: src/components/dictionaryConcepts/components/MapType.jsx [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
 const mapType = (props) => {
   const {
-    index, map_type, map_types, updateEventListener, url,
+    index, map_type, map_types, updateEventListener, url, source,
   } = props;
   return (
     <select
       tabIndex={index}
-      defaultValue={map_type}
+      value={source !== INTERNAL_MAPPING_DEFAULT_SOURCE ? 'NARROWER-THAN' : map_type}
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/components/MapType.jsx]: Shouldn't we use a constant for the above?

File: src/components/dictionaryConcepts/components/MapType.jsx (hunk 2 above)
Comment (judeinno) [reply src/components/dictionaryConcepts/components/MapType.jsx]: I have fixed it.

File: src/tests/Signup/index.test.jsx [hunk 4]
```diff
@@ -28,4 +28,60 @@ describe('Sign up Component', () => {
     wrapper.find('#firstName').simulate('change', event);
     expect(spy).toHaveBeenCalledTimes(1);
   });
+
+  it('should handle text lastname input', () => {
+    const event = {
```
Comment (dkayiwa) [inline src/tests/Signup/index.test.jsx]: How is this test related to the ticket?

Comment (judeinno) [reply src/tests/Signup/index.test.jsx]: Test coverage had dropped and when I checked where the drop was coming from it showed that the drop was coming from it showed the drop was arising from Sign up as shown in the screenshot below so I worked on raising the coverage
image
